packages/tablev2/Table.tsx

```diff
@@ -0,0 +1,168 @@
+  // -- WIDTH --
+  const cell = React.useRef<HTMLDivElement>(null);
+  const [width, setWidth] = React.useState(0);
+  const onRelativeXChange = (change: number) => {
+    const newWidth = Math.max(width + change, 20);
+    state.columns.find(c => c.id === id)!.width = `${newWidth}px`;
+    update({ columns: state.columns.slice(0) });
+  };
+  React.useEffect(() => setWidth(cell.current?.clientWidth || 0));
+
+  // -- SORTING --
+  const order = state.sortBy === id ? state.order : null;
+  const sortClass = config.sorter && style.sortable(order);
+  const nextDir = config.sorter ? (order === "asc" ? "desc" : "asc") : "asc";
+  const onClick =
+    config.sorter && (() => update({ sortBy: id, order: nextDir }));
+
+  return (
+    <div className={style.th} ref={cell} onClick={onClick} role="button">
+      <div className={sortClass}>{config.header}</div>
+      <Draggable onRelativeXChange={onRelativeXChange} />
+    </div>
+  );
+}
+
+type HeaderRowProps = {
+  config: Config;
+  state: State;
+  update: (a: Partial<State>) => void;
+};
+function HeaderRow({ config, state, update }: HeaderRowProps) {
+  const toHeaderCell = ({ id }) => {
+    return (
+      <HeaderCell
+        key={id}
+        id={id}
+        config={config.columns.find(c => c.id === id)!}
+        update={update}
+        state={state}
+      />
+    );
+  };
+  return <>{state.columns.map(toHeaderCell)}</>;
+}
+
+type DivProps = React.HTMLAttributes<HTMLDivElement>;
+export function Table<Entry, ColIDs extends string, Sort extends ColIDs>({
+  data,
+  config,
+  ...divProps
+}: { data: Entry[]; config: Config<Entry, ColIDs, Sort> } & DivProps) {
+  const [state, setState] = React.useState<State>({
+    columns: config.columns.map(c => ({
+      id: c.id,
+      width: c.initialWidth || "max-content"
+    })),
+    order: config.initialSorter?.order || "asc",
+    sortBy: config.initialSorter?.by || null
+  });
+
+  const tableStyle = style.table(
+    state.columns.map(c => c.width),
+    config.stickyFirstCol ?? true
+  );
+
+  const update = (x: Partial<State>) => {
+    setState({ ...state, ...x });
+    config.onStateChange?.({ ...state, ...x });
+  };
+  const col = config.columns.find(c => c.id === state.sortBy);
+  const sort = col?.sorter?.(state.order) || (() => 0);
+
+  // We currently need to cast Config to Config<unknown> as to a bug in typescript's handling of unknowns. We actually NARROW the interface, which is good.
+  return (
+    <div className={tableStyle} {...divProps}>
+      <HeaderRow config={config as Config} state={state} update={update} />
+
+      {data.sort(sort).map(el => (
+        <Row key={"row-" + config.toId(el)} config={config as Config} el={el} />
+      ))}
+    </div>
+  );
+}
+
+const Row = React.memo<{ config: Config; el: unknown }>(({ config, el }) => (
+  <>
+    {config.columns.map(({ id, render }) => (
+      <div key={config.toId(el) + id} className={style.cell}>
```

Comment:
We could improve assistive technology UX by setting `aria-role="gridcell"` and selecting associating each cell with a column header via `aria-describedby`

Reply:
thanks for making that research! we should totally do that!